### ISSUE TITLE
Revert PCRE2.cmake to use Autotools

### ISSUE THREAD

**Issue by cedrik-fuoco-adsk** [AcademySoftwareFoundation/OpenRV#464]
### Discussed in https://github.com/AcademySoftwareFoundation/OpenRV/discussions/462

<div type='discussions-op-text'>

<sup>Originally posted by **keqingMiHoYo** May 12, 2024</sup>
Hello, I encountered many errors regarding stdio.h after I started building via `cmake --build _build --config Release -v --parallel=28 --target main_executable > buildlog.log`. 

Here is my complete log: [buildlog.log](https://github.com/AcademySoftwareFoundation/OpenRV/files/15287147/buildlog.log)
Below is part of my log.

```
    36>CustomBuild:
         	cl  /Zi /Fdossl_static.pdb /Gs0 /GF /Gy /MD /W3 /wd4090 /nologo /O2 /I "." /I "include" -D"L_ENDIAN" -D"OPENSSL_PIC" -D"OPENSSLDIR=\"E:\\GitHub\\OpenRV\\_build\\RV_DEPS_OPENSSL\\install\"" -D"ENGINESDIR=\"E:\\GitHub\\OpenRV\\_build\\RV_DEPS_OPENSSL\\install\\lib\\engines-1_1\"" -D"OPENSSL_SYS_WIN32" -D"WIN32_LEAN_AND_MEAN" -D"UNICODE" -D"_UNICODE" -D"_CRT_SECURE_NO_DEPRECATE" -D"_WINSOCK_DEPRECATED_NO_WARNINGS" -D"OPENSSL_USE_APPLINK" -D"NDEBUG"  -c /Focrypto\objects\obj_dat.obj "crypto\objects\obj_dat.c"
         obj_dat.c
    18>CustomBuild:
           pcre2_auto_possess.c
    36>crypto\objects\obj_dat.c(420): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_OPENSSL.vcxproj]
    36>crypto\objects\obj_dat.c(505): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_OPENSSL.vcxproj]
    36>crypto\objects\obj_dat.c(526): warning C4267: '=': conversion from 'size_t' to 'int', possible loss of data [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_OPENSSL.vcxproj]
    71>CustomBuild:
         	cl.exe /EHsc /MP /MD /I. /DWIN32 /O2 /W0 /nologo        /DLIBRAW_BUILDLIB /Fo"object\\x3f_utils_patched.obj" /c src\x3f\x3f_utils_patched.cpp
    18>C:\msys64\mingw64\include\stdio.h(592,7): error C2059: syntax error: 'type' [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdio.h(599,7): error C2059: syntax error: 'type' [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdio.h(605,7): error C2059: syntax error: 'type' [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdio.h(897,7): error C2059: syntax error: 'type' [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdio.h(929,5): error C2059: syntax error: 'type' [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdio.h(1276,7): error C2059: syntax error: 'type' [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdio.h(1283,7): error C2059: syntax error: 'type' [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdio.h(1289,7): error C2059: syntax error: 'type' [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdio.h(1396,7): error C2059: syntax error: 'type' [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdio.h(1406,7): error C2059: syntax error: 'type' [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\swprintf.inl(14,5): error C2059: syntax error: 'type' [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\swprintf.inl(21,5): error C2059: syntax error: 'type' [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(389,32): error C2085: '_exit': not in formal parameter list [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(389,49): error C2143: syntax error: missing ';' before 'type' [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(396,16): error C2085: '_Exit': not in formal parameter list [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(396,27): error C2143: syntax error: missing ';' before 'type' [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(398,54): error C2085: '_Exit': not in formal parameter list [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(399,3): error C2143: syntax error: missing ';' before '{' [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(399,6): warning C4013: '_exit' undefined; assuming extern returning int [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(399,18): error C2065: 'status': undeclared identifier [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(405,40): error C2059: syntax error: 'type' [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(686,95): error C2491: '_beep': definition of dllimport function not allowed [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(688,24): error C2085: '_seterrormode': not in formal parameter list [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(688,49): error C2143: syntax error: missing ';' before 'type' [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(689,24): error C2085: '_sleep': not in formal parameter list [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(689,56): error C2143: syntax error: missing ';' before 'type' [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(710,17): error C2085: 'ecvt': not in formal parameter list [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(711,17): error C2085: 'fcvt': not in formal parameter list [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(712,17): error C2085: 'gcvt': not in formal parameter list [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(713,17): error C2085: 'itoa': not in formal parameter list [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(714,17): error C2085: 'ltoa': not in formal parameter list [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(715,15): error C2085: 'putenv': not in formal parameter list [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(719,16): error C2085: 'swab': not in formal parameter list [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(722,17): error C2085: 'ultoa': not in formal parameter list [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(723,20): error C2085: 'onexit': not in formal parameter list [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(729,61): error C2085: 'lldiv_t': not in formal parameter list [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(731,21): error C2061: syntax error: identifier 'lldiv_t' [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\stdlib.h(735,72): error C2169: 'llabs': intrinsic function, cannot be defined [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\malloc.h(128,49): error C2065: '_ALLOCA_S_MARKER_SIZE': undeclared identifier [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
    18>C:\msys64\mingw64\include\malloc.h(147,55): error C2065: '_ALLOCA_S_MARKER_SIZE': undeclared identifier [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file '../src/src/pcre2_auto_possess.c')
           
           pcre2_chartables.c
       CustomBuild:
         x3f_utils_patched.cpp
    18>C:\msys64\mingw64\include\stdio.h(592,7): error C2059: syntax error: 'type' [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file 'pcre2_chartables.c')
           
    18>C:\msys64\mingw64\include\stdio.h(599,7): error C2059: syntax error: 'type' [E:\GitHub\OpenRV\_build\RV_DEPS_PCRE2\build\pcre2-8-shared.vcxproj] [E:\GitHub\OpenRV\_build\cmake\dependencies\RV_DEPS_PCRE2.vcxproj]
           (compiling source file 'pcre2_chartables.c')
```</div>

**Comment by cedrik-fuoco-adsk** [comment]
fixed by #465

**Comment by cedrik-fuoco-adsk** [comment]
Fixed by #465
